docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+only have a single GPU-per-node setup:
+
+..
+   note:
+
+   Refer to line 187 in ``model_def_distributed.py``
+
+.. code::
+
+   # NEW: Change selected device to the one with index of local_rank.
+   device = torch.device(core_context.distributed.local_rank)
+
+Shard the data into ``num_replicas`` non-overlapping parts. ``num_replicas`` is equal to
+\`core_context.distributed.size, or the number of slots):
+
+.. code::
+
+   # NEW: Create DistributedSampler object for sharding data into core_context.distributed.size parts.
+   sampler1 = DistributedSampler(
+       dataset1,
+       num_replicas=core_context.distributed.size,
+       rank=core_context.distributed.rank,
+       shuffle=True,
+   )
+   sampler2 = DistributedSampler(
+       dataset2,
+       num_replicas=core_context.distributed.size,
+       rank=core_context.distributed.rank,
+       shuffle=True,
+   )
+
+   # NEW: Shard data.
+   train_loader = torch.utils.data.DataLoader(dataset1, sampler=sampler1, **train_kwargs)
+   test_loader = torch.utils.data.DataLoader(dataset2, sampler=sampler2, **test_kwargs)
+
+Wrap your model with torch’s DistributedDataParallel:
+
+.. code::
+
+   # NEW: Wrap model with DDP. Aggregates gradients and synchronizes model training across slots.
+   model = DDP(model, device_ids=[device], output_device=device)
+
+Finally, at each place in the code where you upload checkpoints, report training metrics, or report
+progress to the master, make sure this is done only on rank 0, e.g.,:
+
+.. code::
+
+   # NEW: Report metrics only on rank 0: only the chief worker may report training metrics and progress,
+   # or upload checkpoints.
+   if core_context.distributed.rank == 0:
+       core_context.train.report_training_metrics(
+           steps_completed=(batch_idx + 1) + (epoch - 1) * len(train_loader),
+           metrics={“train_loss”: loss.item()},
+       )
+
+Step 5.3: Run the Experiment
+============================
+
+Run the following command to run the experiment:
+
+.. code::
+
+   det e create distributed.yaml .
+
+In the Determined WebUI, visit the **Cluster** tab.
+
+You should be able to see multiple slots active in the Cluster tab corresponding to the value you
+set for slots_per_trial you set in distributed.yaml, as well as logs appearing from multiple ranks.
+
+************
+ Next Steps
+************
+
+In this user guide, you learned how to use the Core API to integrate a model into Determined. You
+also saw how to modify a training script and use the appropriate configuration file to report
+metrics and checkpointing, perform a hyperparameter search, and run distributed training.
+
+For more information about distributed training, visit :doc:`Intro to Distributed Training
+</training/dtrain-introduction>`.
+
+..
+   modify the core api mnist blog post part 2 to link to the new core api mnist
+   user guide instead of
+   https://www.determined.ai/blog/intro-to-core-api-pt2
```

Comment:
this looks like a todo comment.  So I'm highlighting it as a reminder.

Reply:
i'll remove this comment when i replace the contents of core-api-ug with the mnist content